pkg/component/etcd/serviceaccount/values_helper.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceaccount
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/pkg/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GenerateValues generates `serviceaccount.Values` for the serviceaccount component with the given `etcd` object.
+func GenerateValues(etcd *druidv1alpha1.Etcd, disableEtcdServiceAccountAutomount bool) *Value {
```

Comment:
I suggest create values.go file. Define only those values there, which is necessary for creating service account. use values_helper.go to intialize the values. The create serviceAccount resource in serviceaccount.go file using those values. Forming serviceaccount struct in values_helper.go is somewhat misleading and does not align with how other resources are deployed. WDYT?

Reply:
Thanks @abdasgupta fixed.